extension-manifest-v3/src/background/reporting.js

```diff
@@ -216,23 +216,41 @@ const reporting = new Reporting({
   communication,
 });
 
-observe('terms', (terms) => {
+function delay(timeInMs) {
+  return new Promise((resolve) => setTimeout(resolve, timeInMs));
+}
+
+function defer() {
+  const deferred = {};
+  deferred.promise = new Promise((resolver) => {
+    deferred.resolve = resolver;
+  });
+  deferred.waitUntil = async ({ timeout, errorMessage = '' } = {}) =>
```

Comment:
Empty string is not a good default value (even if it is not used here), since you may end up with `Uncaught (in promise) <empty string>`. Maybe something like `Timeout exceeded` or leave out the default and the caller always pass it.